tests/TestMembershipContextFactory.php

```diff
@@ -4,59 +4,79 @@
 
 namespace WMDE\Fundraising\MembershipContext\Tests;
 
+use Doctrine\Common\Annotations\AnnotationReader;
 use Doctrine\Common\Annotations\AnnotationRegistry;
 use Doctrine\Common\EventManager;
 use Doctrine\DBAL\Connection;
 use Doctrine\DBAL\DriverManager;
 use Doctrine\ORM\Configuration;
 use Doctrine\ORM\EntityManager;
+use Doctrine\ORM\Tools\Setup;
+use Gedmo\Timestampable\TimestampableListener;
 use WMDE\Fundraising\MembershipContext\Authorization\MembershipTokenGenerator;
 use WMDE\Fundraising\MembershipContext\MembershipContextFactory;
 
 class TestMembershipContextFactory {
 
 	private array $config;
 	private Configuration $doctrineConfig;
-	private Connection $connection;
+	private ?Connection $connection;
 	private MembershipContextFactory $factory;
 	private ?EntityManager $entityManager;
 
-	public function __construct( array $config, Configuration $doctrineConfig ) {
+	public function __construct( array $config ) {
 		$this->config = $config;
-		$this->doctrineConfig = $doctrineConfig;
-
-		$this->connection = DriverManager::getConnection( $this->config['db'] );
-		$this->factory = new MembershipContextFactory( $config, $doctrineConfig );
+		$this->doctrineConfig = Setup::createConfiguration( true );
```

Comment:
I'd prefer if you used `ORMSetup::createXMLMetadataConfiguration`

See https://github.com/wmde/fundraising-memberships/pull/95/files#diff-6e384aac4ab138e655a0e76f12a48b57449642102479088a8aa080d03eaec8d9